pkg/reconciler/identity/reconciler.go

```diff
@@ -126,20 +148,39 @@ func (i *Identity) DeleteWorkloadIdentity(ctx context.Context, projectID string,
 	}
 	if kServiceAccount != nil && len(kServiceAccount.OwnerReferences) == 1 {
 		logging.FromContext(ctx).Desugar().Debug("Removing iam policy binding.")
-		if err := i.removeIamPolicyBinding(ctx, projectID, identifiable.IdentitySpec().GoogleServiceAccount, kServiceAccount); err != nil {
+		if err := i.removeIamPolicyBinding(ctx, projectID, googleServiceAccount, kServiceAccount); err != nil {
 			status.MarkWorkloadIdentityFailed(identifiable.ConditionSet(), deleteWorkloadIdentityFailed, err.Error())
 			return fmt.Errorf("removing iam policy binding failed with: %w", err)
 		}
 	}
 	return nil
 }
 
-func (i *Identity) createServiceAccount(ctx context.Context, namespace, gServiceAccount, clusterName string) (*corev1.ServiceAccount, error) {
-	kServiceAccountName := resources.GenerateServiceAccountName(gServiceAccount, clusterName)
+// getGoogleServiceAccountName will return Google service account name and corresponding raw Kubernetes service account name.
+func (i *Identity) getGoogleServiceAccountName(ctx context.Context, identifiable duck.Identifiable, clusterName string) (string, string, error) {
+	namespace := identifiable.GetObjectMeta().GetNamespace()
+	if identifiable.IdentitySpec().GoogleServiceAccount != "" {
+		gooleServiceAccount := identifiable.IdentitySpec().GoogleServiceAccount
+		return gooleServiceAccount, resources.GenerateServiceAccountName(gooleServiceAccount, clusterName), nil
+	}
+	cm, err := i.kubeClient.CoreV1().ConfigMaps("cloud-run-events").Get(gcpauth.ConfigMapName(), metav1.GetOptions{})
```

Comment:
The Identity struct should have a `gcpauth.Store` field, created when the controller is created. Then here we use:

```golang
ad := i.gcpAuthStore.Load()
if ad == nil || ad.GCPAuthDefaults == nil {
  error
}
gsa := ad.GCPAuthDefaults.WorkloadIdentityGSA(namespace, ksa)
```